provider/pkg/custom/interface.go

```diff
@@ -0,0 +1,18 @@
+package custom
+
+import (
+	"context"
+
+	"github.com/pulumi/pulumi/sdk/v3/go/common/resource"
+)
+
+type CustomResource interface {
+	// Create creates a new resource in the cloud provider and returns its unique identifier and outputs.
+	Create(ctx context.Context, urn resource.URN, inputs resource.PropertyMap) (identifier *string, outputs resource.PropertyMap, err error)
+	// Read returns the outputs and the updated inputs of the resource.
+	Read(ctx context.Context, urn resource.URN, id string, oldInputs, oldOutputs resource.PropertyMap) (resource.PropertyMap, resource.PropertyMap, bool, error)
```

Comment:
Passing a `urn` to these methods feels odd to me. Isn't it on a wrong level?

I think you only use it once in a logging statement, I commented on its usage specifically.

Reply:
Yes, it's for logging which a custom resource might well want to do during any given operation.